diff --git a/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero.dtsi b/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero.dtsi
index 3ab053bb0f..3d98d21cbd 100644
--- a/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero.dtsi
+++ b/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero.dtsi
@@ -68,6 +68,43 @@
 		enable-active-high;
 		gpio = <&pio 2 16 GPIO_ACTIVE_HIGH>; /* PC16 */
 	};
+
+
+	reg_vcc3v3: vcc3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-3v3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+
+	reg_vcc33_wifi: vcc33-wifi {
+		/* Always on 3.3V regulator for WiFi and BT */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc33-wifi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+		vin-supply = <&reg_vcc5v>;
+	};
+
+	reg_vcc_wifi_io: vcc-wifi-io {
+		/* Always on 1.8V/300mA regulator for WiFi and BT IO */
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-wifi-io";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		regulator-always-on;
+		vin-supply = <&reg_vcc33_wifi>;
+	};
+
+	wifi_pwrseq: wifi-pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		clocks = <&rtc 1>;
+		clock-names = "osc32k-out";
+		reset-gpios = <&pio 6 18 GPIO_ACTIVE_LOW>; /* PG18 */
+		post-power-on-delay-ms = <200>;
+	};
 };
 
 &ehci1 {
diff --git a/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero2.dts b/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero2.dts
index 0e2d1f1629..596ad08045 100644
--- a/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero2.dts
+++ b/dts/upstream/src/arm64/allwinner/sun50i-h616-orangepi-zero2.dts
@@ -30,7 +30,17 @@
 };
 
 &mmc0 {
-	vmmc-supply = <&reg_dcdce>;
+	vmmc-supply = <&reg_vcc3v3>;
+};
+
+&mmc1 {
+	vmmc-supply = <&reg_vcc33_wifi>;
+	vqmmc-supply = <&reg_vcc_wifi_io>;
+	mmc-pwrseq = <&wifi_pwrseq>;
+	bus-width = <4>;
+	non-removable;
+	mmc-ddr-1_8v;
+	status = "okay";
 };
 
 &r_i2c {
@@ -191,9 +201,9 @@
 };
 
 &pio {
-	vcc-pc-supply = <&reg_aldo1>;
-	vcc-pf-supply = <&reg_aldo1>;
-	vcc-pg-supply = <&reg_bldo1>;
-	vcc-ph-supply = <&reg_aldo1>;
-	vcc-pi-supply = <&reg_aldo1>;
+	vcc-pc-supply = <&reg_vcc3v3>;
+	vcc-pf-supply = <&reg_vcc3v3>;
+	vcc-pg-supply = <&reg_vcc3v3>;
+	vcc-ph-supply = <&reg_vcc3v3>;
+	vcc-pi-supply = <&reg_vcc3v3>;
 };
