diff --git a/arch/arm/dts/sun50i-h616-orangepi-zero.dtsi b/arch/arm/dts/sun50i-h616-orangepi-zero.dtsi
index 3d98d21cbd..da101398d3 100644
--- a/arch/arm/dts/sun50i-h616-orangepi-zero.dtsi
+++ b/arch/arm/dts/sun50i-h616-orangepi-zero.dtsi
@@ -45,7 +45,7 @@
 
 		led-1 {
 			function = LED_FUNCTION_STATUS;
-			color = <LED_COLOR_ID_GREEN>;
+			color = <LED_COLOR_ID_BLUE>;
 			gpios = <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
 		};
 	};
